### DISCUSSION TITLE
Event de-duplication

### DISCUSSION THREAD

**Discussion by dualarrow** [exceptionless/Exceptionless#1087] [Q&A] [answered]
In the documentation its mentioned that the client does de-duplication of identical events and that it does this by holding the 2nd and any further identical events over a 60 second period and increments the count. Theres no mention that I could find on how a custom client would pass this count on to the server.

Is it just a field in the JSON, or a header and if so are you able to provide some details so I could implement it on my custom client.

Also, for usage stats, you send a message with a type of "Usage" but where do you put the usage name ? I have tried in the message and in a tag but all entries seem to be bundled together.

**Comment by dualarrow** [comment]
I found the usage problem I was having. The type needs to be "Usage" as "usage" doesnt seem to work. Also, you need to specify a source with the name of the feature.

Still haven't found a solution for the de-dupe

**Reply by ejsmith** [reply]
De-dupe is a purely client side thing. The client is storing a hash signature of each event as it's being submitting and checking against that to avoid sending the exact same event.

**Comment by niemyjski** [comment, marked as answer]
1. It sets the usage in the count event property: https://github.com/exceptionless/Exceptionless.Net/blob/65cc5353738cef51f3e8e273dc6d6f71c6f29a07/src/Exceptionless/Plugins/Default/910_DuplicateCheckerPlugin.cs#L114
2.  Set the feature usage name in the source property. We recommend this be a valid identifier.

Please let us know if you have any other questions.
